### PR TITLE
1467: move vt version number into a separate file

### LINKED ISSUE #1467: move `vt` version number into a separate file
Currently every header file in `vt` contains version number like this:
```cpp
//                           DARMA Toolkit v. 1.0.0
```
This is currently not updated after a new release and could pollute the git history when we have frequent releases.

**What Needs to be Done?**
- [x] remove version number from headers
- [x] store it in a single place (`VERSION` file)
- [x] make sure that `find_package(vt)` works correctly

Additional preparation for releases based on `develop`:
- [x] ~describe versioning scheme (`1.1M6 (‘Milestone 6’)` vs `1.1.6`) on wiki (optional: check Sandia requirements for this)~ sticking with semver for now
- [ ] describe new (weekly) release process on wiki
- [x] create a `main` branch (or use `master`?)
- [ ] setup permissions and builds for `main`

Sidenote: as far as release notes go, there is ~3,5k commits added and ~350 issues resolved on `develop` (compared to `1.0.2` release):
```
vt  git log --oneline  master..develop | wc -l
3556
```
```
vt  git log --oneline  master..develop | grep "Merge pull request" | cut -d# -f2- | cut -c 1-4 | sort -u | wc -l
353
```

<details>
  <summary>list of the issues</summary>

#454 
#461 
#463 
#464 
#473 
#479 
#481 
#483 
#485 
#487 
#488 
#495 
#499 
#502 
#504 
#505 
#508 
#513 
#514 
#516 
#518 
#520 
#525 
#528 
#542 
#543 
#544 
#545 
#546 
#547 
#549 
#550 
#555 
#557 
#562 
#565 
#569 
#571 
#573 
#576 
#578 
#579 
#583 
#588 
#590 
#593 
#594 
#598 
#606 
#609 
#613 
#614 
#616 
#618 
#622 
#624 
#626 
#629 
#632 
#638 
#640 
#642 
#644 
#648 
#650 
#652 
#654 
#658 
#661 
#662 
#664 
#669 
#670 
#674 
#675 
#678 
#681 
#683 
#690 
#692 
#693 
#698 
#700 
#701 
#704 
#711 
#714 
#718 
#720 
#722 
#724 
#725 
#727 
#730 
#731 
#733 
#735 
#737 
#739 
#743 
#745 
#747 
#748 
#751 
#753 
#755 
#757 
#766 
#768 
#770 
#773 
#775 
#777 
#779 
#782 
#783 
#784 
#787 
#790 
#792 
#795 
#798 
#801 
#804 
#805 
#809 
#812 
#815 
#818 
#824 
#826 
#828 
#832 
#834 
#836 
#838 
#843 
#845 
#849 
#852 
#855 
#857 
#859 
#862 
#863 
#864 
#866 
#887 
#890 
#892 
#894 
#897 
#898 
#899 
#901 
#902 
#906 
#907 
#915 
#921 
#923 
#924 
#928 
#932 
#933 
#934 
#936 
#938 
#944 
#945 
#947 
#952 
#953 
#957 
#961 
#963 
#968 
#969 
#972 
#974 
#975 
#977 
#979 
#980 
#982 
#983 
#985 
#989 
#993 
#994 
#995 
#996 
#997 
#998 
#1001
#1008
#1010
#1011
#1012
#1013
#1019
#1026
#1027
#1028
#1029
#1032
#1035
#1038
#1041
#1043
#1046
#1057
#1058
#1064
#1065
#1067
#1068
#1076
#1077
#1078
#1081
#1082
#1084
#1085
#1087
#1089
#1091
#1093
#1095
#1101
#1102
#1103
#1104
#1106
#1107
#1108
#1109
#1113
#1118
#1119
#1121
#1123
#1130
#1132
#1133
#1135
#1137
#1138
#1143
#1145
#1146
#1152
#1155
#1160
#1164
#1165
#1167
#1170
#1171
#1178
#1179
#1181
#1182
#1185
#1188
#1189
#1191
#1196
#1200
#1208
#1209
#1210
#1214
#1220
#1221
#1223
#1225
#1226
#1228
#1230
#1236
#1238
#1241
#1245
#1246
#1251
#1252
#1253
#1257
#1259
#1262
#1263
#1264
#1268
#1269
#1271
#1280
#1285
#1287
#1288
#1293
#1299
#1301
#1305
#1306
#1310
#1319
#1321
#1322
#1324
#1325
#1327
#1329
#1330
#1332
#1335
#1343
#1346
#1348
#1354
#1356
#1360
#1365
#1370
#1371
#1373
#1374
#1380
#1384
#1388
#1391
#1400
#1403
#1404
#1405
#1411
#1412
#1414
#1415
#1421
#1422
#1424
#1429
#1431
#1434
#1440
#1444
#1448
#1455
#1457
#1459
#1461
#1464

</details>

### PR DESCRIPTION
**Pull request by cz4rs**
fixes #1467

### CODE REVIEW

**Comment by cz4rs** [comment]
Most of the changes boil down to:
```
sed -i "/DARMA Toolkit/d"
```

Other files affected are at the top of the diff:
```
 CMakeLists.txt                                                        | 2 +-
 LICENSE                                                               | 2 +-
 VERSION                                                               | 1 +
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 cmake_minimum_required(VERSION 3.17 FATAL_ERROR)
-project(vt)
+project(vt VERSION 1.0.2)
```
Comment (nlslatt) [inline CMakeLists.txt:2]: Given how much `develop` differs from the 1.0.2 release, is this really the right version number to use here?

Comment (cz4rs) [reply CMakeLists.txt:2]: I was thinking about this as well, but I haven't come up with the new number.

Comment (cz4rs) [reply CMakeLists.txt:2]: `1.1.0` makes sense using semver, but we might want to move to milestone notation if we are going to stick with that.

Comment (cz4rs) [reply CMakeLists.txt]: changed to 1.1.0, approved at DARMA weekly

File: LICENSE [hunk 2]
```diff
@@ -1,5 +1,5 @@
-                          DARMA Toolkit v. 1.0.0
                        DARMA/vt => Virtual Transport
+                    For version number see VERSION file
```
Comment (cz4rs) [inline LICENSE:2]: `DARMA Toolkit v. 1.0.0` might need to remain here

**Review by nlslatt — APPROVED** [review]

**Pull request merged**